openslides/utils/class_register.py

```diff
@@ -0,0 +1,39 @@
+class RegisterSubclasses(type):
+    """
+    Metaclass to track subclasses of a class.
+
+    The metaclass adds an attribute 'subclasses' to the metaclass. This
+    attribute is a set containing all derivated classes.
+
+    After a new subclass is created, the classmethod on_subclass_created is
+    called.
+
+    Example:
+    class Base(metaclass=RegisterSubclasses):
+        pass
+
+    class Child(Base):
+        pass
+
+    assert Child in Base.subclasses
```

Comment:
```python
class Me(metaclass=RegisterSubclasses):
    pass

class ChildOne(Me):
    pass

class ChildTwo(Me):
    pass

class Grandchild(ChildOne):
    pass
```
Now it is missleading if `ChildOne.subclasses` returns a set containing also ChildTwo. Also it is missleading that 
`Grandchild.subclasses` contains the whole family but not the grandfather.

May be you want to restrict the subclasses attribute to be available only for the base class may be by deleting the reference or something like that.